server/src/test/java/org/apache/druid/client/DirectDruidClientTest.java

```diff
@@ -427,4 +431,47 @@ public void testQueryTimeoutFromFuture()
     Assert.assertEquals(hostName, actualException.getHost());
     EasyMock.verify(httpClient);
   }
+
+  @Test
+  public void testRuntimeException() throws JsonProcessingException
```

Comment:
This is too generic test name. I would suggest testConnectionCountAfterException